### PR TITLE
2D Palette swap shader

### LINKED ISSUE #13: indexing color, palette swap, indexed palette swapping
**I'm submitting a...**

- [ ] Bug report.
- [x] Feature request.

## Feature request ##

**Describe the problem you're trying to solve.**
 it would be nice to have the following shaders..
- **Indexed Color** allows you to limit every pixel on screen to a palette.
- **Palette Swap** allows you to swap colors that are on screen for different colors.
- **Indexed Palette Swap** indexes the colors to the original palette and then swaps them depending on order.

I would just suggest that it be made a shader

**Tell us which solutions you've explored, the solution you would pick, and why you think it would be the best for everyone.**
I am currently trying to get help on the discord to implement a Indexed Color shader but it might be nice to beable to swap colors later

I think it would be good for retro games regardless if they are 2d or 3d

### PR DESCRIPTION
**Pull request by nezvers**
Closes #13

### CODE REVIEW

File: godot/Demos/PaletteSwap2D/PaletteSwap2D.gd [hunk 1]
```diff
@@ -0,0 +1,14 @@
+tool
+extends Control
+
+export (int) var index: = 0 setget set_index
+var palette_count: = 4
+onready var sprite:= $Sprite_code_shader
+
+func set_index(value:int)->void:
+	index = value % palette_count								#loop index between palette count
```
Comment (NathanLovato) [inline godot/Demos/PaletteSwap2D/PaletteSwap2D.gd:9]: Could you please remove inline code comments?

File: godot/Shaders/PalettSwap2D.shader [hunk 2]
```diff
@@ -0,0 +1,18 @@
+shader_type canvas_item;
+render_mode unshaded, blend_disabled;
+
+uniform sampler2D palette;									//Use palletes in collum and color set horizontaly
```
Comment (NathanLovato) [inline godot/Shaders/PalettSwap2D.shader:4]: Could you remove comments in this file?

File: godot/Demos/PaletteSwap2D.tscn [hunk 3]
```diff
@@ -0,0 +1,65 @@
+[gd_scene load_steps=8 format=2]
```
Comment (NathanLovato) [inline godot/Demos/PaletteSwap2D.tscn:1]: The file name and its root node should be PaletteSwapDemo to be consistent with the rest of the demos.
Same for the GDScript file.

**Comment by nezvers** [comment]
Ok, I did that in my fork, What do I do next?

**Comment by NathanLovato** [comment]
It's all good. The code needs a tiny bit of formatting but I'll take care of that. Thanks for your contribution!

**Pull request merged**